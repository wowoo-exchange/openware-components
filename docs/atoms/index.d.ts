export * from './Avatar/Avatar';
export * from './Button/Button';
export * from './Checkbox/Checkbox';
export * from './CryptoIcon/CryptoIcon';
export * from './Input/Input';
export * from './Loader/Loader';
export * from './HeaderItem/HeaderItem';
export * from './Alert/Alert';
export * from './AlertIcon/AlertIcon';
export * from './Pagination/Pagination';
export * from './CloseButton/CloseButton';
export * from './PercentageButton/PercentageButton';
export * from './SummaryField/SummaryField';
