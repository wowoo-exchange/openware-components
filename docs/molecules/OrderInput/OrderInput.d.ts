import * as React from 'react';
interface OrderInputProps {
    /**
     * Additional class name for styling. By default element receives `cr-input-block` class
     * @default empty
     */
    className?: string;
    /**
     * Code of cryptocurrency
     * @default empty
     */
    currency: string;
    /**
     * Checking if input focused
     */
    isFocused: boolean;
    /**
     * Label on the border
     */
    label?: string;
    /**
     * Placeholder on the border
     */
    placeholder?: string;
    /**
     * Value of Input component
     */
    value: string | number;
    /**
     * Function for getting value of input
     * @default empty
     */
    handleChangeValue: (text: string) => void;
    /**
     * Function for handling input focus
     */
    handleFocusInput: () => void;
    /**
     * Function for handling 'Enter' key
     */
    onKeyPress?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
}
/**
 * Input with cryptocurrency icon and label.
 */
declare const OrderInput: React.FunctionComponent<OrderInputProps>;
export { OrderInput, OrderInputProps, };
