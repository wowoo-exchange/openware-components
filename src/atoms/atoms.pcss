@import "Avatar/Avatar.pcss";
@import "Button/Button.pcss";
@import "Checkbox/Checkbox.pcss";
@import "CryptoIcon/CryptoIcon.pcss";
@import "Input/Input.pcss";
@import "Loader/Loader.pcss";
@import "CloseButton/CloseButton.pcss";
@import "PercentageButton/PercentageButton.pcss";
@import "HeaderItem/HeaderItem.pcss";
@import "Alert/Alert.pcss";
@import "AlertIcon/AlertIcon.pcss";
@import "Pagination/Pagination.pcss";
