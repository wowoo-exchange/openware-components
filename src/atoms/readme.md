Very simple building blocks of matter. These components are independent and highly customizable and canâ€™t be broken down further. 
