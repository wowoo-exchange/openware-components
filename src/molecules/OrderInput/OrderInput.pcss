.cr-order-input {
  display: flex;
  justify-content: space-between;

  &__fieldset {
    background-color: var(--order-input-input-bg);
    border: 1px solid var(--order-input-border-color);
    color: var(--order-input-input-color);
    display: inline-block;
    position: relative;
    margin: 0;
    margin-top: calc(var(--gap) * 2);
    width: 100%;

    &__label {
      background-color: var(--order-input-input-bg);
      position: absolute;
      top: calc(var(--gap) * -1.3);
    }

    &__input {
      background-color: var(--order-input-input-bg);
      border: none;
      margin: var(--gap);
      width: 100%;

      input {
        width: 100%;
      }
    }
  }

  &__crypto-icon {
    background-color: var(--order-input-input-bg);
    border: 1px solid var(--order-input-border-color);
    border-left: none;
    display: inline-block;
    margin-top: calc(var(--gap) * 2);
    padding: calc(var(--gap) * 2);
    text-align: center;
    width: 20%;

    span {
      background-color: inherit;
    }
  }
}
