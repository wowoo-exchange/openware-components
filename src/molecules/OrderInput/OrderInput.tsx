import cr from 'classnames';
import * as React from 'react';
import { CryptoIcon, Input } from '../../atoms';

interface OrderInputProps {
    /**
     * Additional class name for styling. By default element receives `cr-input-block` class
     * @default empty
     */
    className?: string;
    /**
     * Code of cryptocurrency
     * @default empty
     */
    currency: string;
    /**
     * Checking if input focused
     */
    isFocused: boolean;
    /**
     * Label on the border
     */
    label?: string;
    /**
     * Placeholder on the border
     */
    placeholder?: string;
    /**
     * Value of Input component
     */
    value: string | number;
    /**
     * Function for getting value of input
     * @default empty
     */
    handleChangeValue: (text: string) => void;
    /**
     * Function for handling input focus
     */
    handleFocusInput: () => void;
    /**
     * Function for handling 'Enter' key
     */
    onKeyPress?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
}

/**
 * Input with cryptocurrency icon and label.
 */
const OrderInput: React.FunctionComponent<OrderInputProps> = (props: OrderInputProps) => {
    const { currency, className, isFocused, label, placeholder, value, handleChangeValue, handleFocusInput, onKeyPress } = props;
    const cx = cr('cr-order-input', className);
    const currencyCode = `${currency.toUpperCase()}-alt`;

    const fieldsetFocusedClass = cr('cr-order-input__fieldset', {
        'cr-order-input__fieldset cr-order-input__fieldset--focused': isFocused,
    });

    return (
        <div className={cx}>
            <fieldset className={fieldsetFocusedClass}>
                {value && label ? (
                        <legend className={'cr-order-input__fieldset__label'}>
                            {label}
                        </legend>
                    ) :
                null}
                <Input
                    type="text"
                    value={value}
                    placeholder={placeholder ? placeholder : '0'}
                    className="cr-order-input__fieldset__input"
                    onChangeValue={handleChangeValue}
                    onFocus={handleFocusInput}
                    onBlur={handleFocusInput}
                    onKeyPress={onKeyPress}
                />
            </fieldset>
            <div className="cr-order-input__crypto-icon">
                <CryptoIcon code={currencyCode}>{currency.toUpperCase()}</CryptoIcon>
            </div>
        </div>
    );
};

export {
    OrderInput,
    OrderInputProps,
};
