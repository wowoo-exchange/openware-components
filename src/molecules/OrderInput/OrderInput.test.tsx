import { mount, shallow } from 'enzyme';
import * as React from 'react';
import { Input } from '../../atoms';
import { OrderInput, OrderInputProps } from './OrderInput';

const defaultProps: OrderInputProps = {
    currency: 'eth',
    handleChangeValue: () => undefined,
    handleFocusInput: () => undefined,
    isFocused: false,
    value: '',
    onKeyPress: jest.fn(),
};

const setup = (props: Partial<OrderInputProps> = {}) =>
    shallow(<OrderInput {...{ ...defaultProps, ...props }} />);

describe('InputBlock', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('should have correct className', () => {
        const wrapper = setup();
        expect(wrapper.hasClass('cr-order-input')).toBeTruthy();
    });

    it('should pass along supplied className', () => {
        const className = 'some-class';
        const wrapper = setup({ className });
        expect(wrapper.hasClass(className)).toBeTruthy();
    });

    it('should set correct currency code to the child components', () => {
        const wrapper = setup();
        const { children } = wrapper.find('CryptoIcon').props();
        expect(children).toContain('ETH');
    });

    it('inner Input component should render with correct type', () => {
        defaultProps.value = '0.02';
        const change = defaultProps.handleChangeValue;
        const keyPress = defaultProps.onKeyPress;
        const { value } = mount(
            (
                <Input
                    className="cr-input-block__input"
                    type="text"
                    value="0.02"
                    onChangeValue={change}
                    onKeyPress={keyPress}
                />
            )).props();
        expect(value).toContain(defaultProps.value);
    });
});
