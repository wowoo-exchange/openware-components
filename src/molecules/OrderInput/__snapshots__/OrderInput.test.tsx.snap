// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBlock should render 1`] = `
<div
  className="cr-order-input"
>
  <fieldset
    className="cr-order-input__fieldset"
  >
    <Input
      className="cr-order-input__fieldset__input"
      onBlur={[Function]}
      onChangeValue={[Function]}
      onFocus={[Function]}
      onKeyPress={[MockFunction]}
      placeholder="0"
      type="text"
      value=""
    />
  </fieldset>
  <div
    className="cr-order-input__crypto-icon"
  >
    <CryptoIcon
      code="ETH-alt"
    >
      ETH
    </CryptoIcon>
  </div>
</div>
`;
